Code/Framework/AzCore/AzCore/Math/Capsule.h

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Math/Vector3.h>
+
+namespace AZ
+{
+    //! A class representing a capsule (a cylinder with a hemispherical cap at each end).
+    class Capsule
+    {
+    public:
+        AZ_TYPE_INFO(Capsule, "{4B7E9154-B258-40D2-81A4-A27A25030588}");
+
+        Capsule() = default;
+        Capsule(const Vector3& firstHemisphereCenter, const Vector3& secondHemisphereCenter, float radius);
```

Comment:
suggestion: This could totally be in a separate PR, but it might be nice if Capsule also had a constructor for the new `AZ::LineSegment` type, could be a helpful utility to create a capsule from a line segment, but it's just a nice to have, not at all critical